### PR TITLE
[WIP] Inline syntax for single command

### PR DESCRIPTION
**Pull request by IvanShamatov**
https://github.com/dry-rb/dry-cli/issues/63 According to this issue

### CODE REVIEW

File: lib/dry/cli/inline.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      private
+
+      def command_names
+        [ProgramName.call]
+      end
+    end
+  end
+end
+
+include Dry::CLI::Inline
```
Comment (solnic) [inline lib/dry/cli/inline.rb]: Codacy Issue found: [`include` is used at the top level. Use inside `class` or `module`.](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

**Comment by solnic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/dry/cli/inline.rb  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973&bid=15976225)

File: lib/dry/cli.rb [hunk 2]
```diff
@@ -85,14 +107,10 @@ def call(arguments: ARGV, out: $stdout)
...
-    # @since 0.1.0
+    # @since 0.6.x
     # @api private
-    def parse(result, out)
-      command = result.command
-      names = result.names
-      return [command, result.arguments] unless command?(command)
-
-      result = Parser.call(command, result.arguments, result.names)
+    def parse_arguments(command, arguments, names, out)
```
Comment (solnic) [inline lib/dry/cli.rb]: Codacy Issue found: [Use empty lines between method definitions.](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

File: lib/dry/cli.rb [hunk 3]
```diff
@@ -56,10 +59,34 @@ def initialize(registry = nil, &block)
...
+    attr_reader :command
+
+    def perform_command(arguments, out)
+      kommand = command.is_instance_of?(Command) ? command : command.new
+      args = parse_arguments(kommand, arguments, names, out)
+      kommand.call(args)
+    end
+
+
+    def perform_registry(arguments, out)
```
Comment (solnic) [inline lib/dry/cli.rb]: Codacy Issue found: [Use empty lines between method definitions.](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

**Comment by solnic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Coverage decreased per file
===========================
- lib/dry/cli.rb  -5
         

Complexity decreasing per file
==============================
+ lib/dry/cli.rb  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973&bid=15976225)

File: lib/dry/cli.rb [hunk 4]
```diff
@@ -154,11 +172,12 @@ def anonymous_registry(&block)
 
   # Create a new instance
   #
-  # @param registry [Dry::CLI::Registry, Block] a registry or a configuration block
+  # @param registry_or_command [Dry::CLI::Registry, Dry::CLI::Command] a registry or singular command
```
Comment (solnic) [inline lib/dry/cli.rb]: Codacy Issue found: [Line is too long. [101/100]](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

File: lib/dry/cli/inline.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    require 'dry/cli'
+
+    module Inline
+      extend Forwardable
+
+      AnonymousCommand = Class.new(Dry::CLI::Command)
+
+      delegate %i[desc example argument option] => AnonymousCommand
+
+      def run(arguments: ARGV, out: $stdout)
```
Comment (solnic) [inline lib/dry/cli/inline.rb]: Codacy Issue found: [Unused method argument - `out`. You can also write as `run(*)` if you want the method to accept any arguments but don't care about them.](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

File: lib/dry/cli/inline.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+          yield(*args)
+        end
+
+        Dry.CLI(command).call(arguments: ARGV, out: $stdout)
+      end
+    end
+  end
+end
+
+include Dry::CLI::Inline
```
Comment (solnic) [inline lib/dry/cli/inline.rb]: Codacy Issue found: [`include` is used at the top level. Use inside `class` or `module`.](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

File: lib/dry/cli.rb [hunk 7]
```diff
@@ -36,16 +36,19 @@ def self.command?(command)
 
     # Create a new instance
     #
-    # @param registry [Dry::CLI::Registry, Block] a registry or a configuration block
+    # @param command_or_registry [Dry::CLI::Registry, Dry::CLI::Command] a registry or singular command
```
Comment (solnic) [inline lib/dry/cli.rb]: Codacy Issue found: [Line is too long. [103/100]](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

**Comment by solnic** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Coverage decreased per file
===========================
- lib/dry/cli.rb  -5
         

Complexity decreasing per file
==============================
+ lib/dry/cli.rb  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973&bid=15976225)

File: lib/dry/cli.rb [hunk 8]
```diff
@@ -154,11 +185,12 @@ def anonymous_registry(&block)
 
   # Create a new instance
   #
-  # @param registry [Dry::CLI::Registry, Block] a registry or a configuration block
+  # @param registry_or_command [Dry::CLI::Registry, Dry::CLI::Command] a registry or singular command
```
Comment (solnic) [inline lib/dry/cli.rb]: Codacy Issue found: [Line is too long. [101/100]](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

File: lib/dry/cli.rb [hunk 9]
```diff
@@ -85,14 +120,10 @@ def call(arguments: ARGV, out: $stdout)
...
-    # @since 0.1.0
+    # @since 0.6.x
     # @api private
-    def parse(result, out)
-      command = result.command
-      names = result.names
-      return [command, result.arguments] unless command?(command)
-
-      result = Parser.call(command, result.arguments, result.names)
+    def parse_arguments(command, arguments, names, out)
```
Comment (solnic) [inline lib/dry/cli.rb]: Codacy Issue found: [Use empty lines between method definitions.](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

File: lib/dry/cli.rb (hunk 7 above)
Comment (solnic) [inline lib/dry/cli.rb]: Codacy Issue found: [Line is too long. [103/100]](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

File: lib/dry/cli/inline.rb [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+          yield(*args)
+        end
+
+        Dry.CLI(command).call(arguments: arguments, out: out)
+      end
+    end
+  end
+end
+
+include Dry::CLI::Inline
```
Comment (solnic) [inline lib/dry/cli/inline.rb]: Codacy Issue found: [`include` is used at the top level. Use inside `class` or `module`.](https://app.codacy.com/gh/dry-rb/dry-cli/pullRequest?prid=4800973)

**Comment by IvanShamatov** [comment]
Closing this one. 
Will do 2 different PR's:
1) with ability to pass Command as well as registry;
2) with inline syntax for running commands.

**Pull request closed without merging**